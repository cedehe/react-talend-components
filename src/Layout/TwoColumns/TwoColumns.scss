////
/// Layout
/// @group Custom widgets
////

/// Layout first columns in two columns background color
/// @type Color
$tc-layout-two-columns-sidemenu-bg: #4F8992 !default;

.container {
	display: flex;
	flex-direction: row;
}

.sidemenu {
	background-color: $tc-layout-two-columns-sidemenu-bg;

	:global(.nav-pills) :global(.btn-link) {
		color: #FFF;
		width: 100%;
	}

	:global(.btn-link):hover {
		background-color: darken($tc-layout-two-columns-sidemenu-bg, 12%);
	}
}

.main {
	flex-grow: 1;
	min-height: 0;
	overflow-y: auto;
}
