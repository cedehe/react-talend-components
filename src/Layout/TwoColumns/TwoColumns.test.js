import React from 'react';
import renderer from 'react-test-renderer';

import TwoColumns from './TwoColumns.component';

describe('TwoColumns', () => {
	it('should render two columns', () => {
		const wrapper = renderer.create(
			<TwoColumns one="Hello world">
				<span>children</span>
			</TwoColumns>).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
