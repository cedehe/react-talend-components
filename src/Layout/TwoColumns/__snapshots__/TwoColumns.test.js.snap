exports[`TwoColumns should render two columns 1`] = `
<div
  className="tc-layout-two-columns">
  <div
    className="tc-layout-two-columns-left">
    Hello world
  </div>
  <div
    className="tc-layout-two-columns-main">
    <span>
      children
    </span>
  </div>
</div>
`;
